common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -584,10 +585,17 @@ protected TokenResult getTokenResultFromHttpResponse(@NonNull final HttpResponse
 
         if (response.getStatusCode() >= HttpURLConnection.HTTP_BAD_REQUEST) {
             //An error occurred
-            tokenErrorResponse = ObjectMapper.deserializeJsonStringToObject(
-                    getBodyFromUnsuccessfulResponse(response.getBody()),
-                    MicrosoftTokenErrorResponse.class
-            );
+            try {
+                tokenErrorResponse = ObjectMapper.deserializeJsonStringToObject(
+                        getBodyFromUnsuccessfulResponse(response.getBody()),
+                        MicrosoftTokenErrorResponse.class
+                );
+            } catch (final JsonParseException ex) {
+                tokenErrorResponse = new MicrosoftTokenErrorResponse();
+                final String statusCode = String.valueOf(response.getStatusCode());
+                tokenErrorResponse.setError(statusCode);
```

Comment:
Looks like this is set same as status code on line 597. Do we have a more specific error code (rather than a status code)?

Reply:
Not sure if there is anything more specific. open to suggestion.